pkg/abap/build/connector.go

```diff
@@ -174,3 +176,39 @@ func (conn Connector) UploadSarFile(appendum string, sarFile []byte) error {
 	defer response.Body.Close()
 	return nil
 }
+
+// UploadSarFileInChunks : upload *.sar file in chunks
+func (conn Connector) UploadSarFileInChunks(appendum string, fileName string, sarFile []byte) error {
+	//Maybe Next Refactoring step to read the file in chunks, too?
+
+	url := conn.Baseurl + appendum
+
+	header := make(map[string][]string)
+	header["Content-Disposition"] = []string{"form-data; name=\"file\"; filename=\"" + fileName + "\""}
+
+	//chunkSize := 10000 // 10KB for testing
```

Comment:
Why not add a private version of this function for testing, which takes the chunk size as parameter. And the public function passes a larger default value to the implementation function.

Reply:
This was actually for real developer / integration / e2e tests. It is not that easy to produce a file which is accepted by the backend in a reasonable size for testing.